### ISSUE TITLE
Testing tray-only (no window) app

### ISSUE THREAD

**Issue by hovancik** [electron/spectron#90]
How would I go about testing tray-only (no windows) app?

Every example I can find uses `app.client.waitUntilWindowLoaded()`, which is obviously no go with tray-only app. There is no such a thing as `waitUntilTrayLoaded()`, as well. 

I am trying to make a simple test here: https://github.com/hovancik/strechly/pull/4/files but I am getting errors: https://travis-ci.org/hovancik/strechly/jobs/145827763 and no idea how to move.

**Comment by kevinsawicki** [comment]
I don't think it is possible at this time to test tray only apps using spectron.

Spectron is built on top of Chromedriver which tests renderer processes. Tray code runs in the main process and so it cannot be tested through Chromedriver unless a renderer process is available to run code in.

**Comment by hovancik** [comment]
@kevinsawicki I see, thanks . 

Anyway, meanwhile I was thinking about how to make a workaround and I made a small one: 
I am showing startup window for 5 seconds and that way I am able to test the start of my app works [here](https://github.com/hovancik/strechly/blob/11cafcfab3f6d53f46b4f44600c977a51abe71d2/tests/index.js) 

I have a problem thought,I am not able to do: 

```
 after(function () {
    if (app && app.isRunning()) {
       return app.stop()
     }
 })
```

As I am getting: 

```
 strechly "after all" hook:
 RuntimeError: no such window: target window already closed
 from unknown error: web view not found
    at window() - application.js:206:19
```

Maybe you happen to know how to end app process? I wasn't able to find anything.

**Comment by kevinsawicki** [comment]
Calling `app.stop()` currently requires at least one created window (hidden or showing) since stopping the app routes through the renderer process.

This could probably be fixed though, I'll take a look.

**Comment by mykter** [comment]
It would be useful if this were mentioned in the docs.
app.start() never returns for apps that don't open a window on startup.
